src/desktop/analytics/helpers.ts

```diff
@@ -6,6 +6,26 @@ import { trackTimeOnPage } from "./timeOnPageListener"
 const setupSplitTests = require("../components/split_test/setup.coffee")
 const Events = require("../../v2/Utils/Events").default
 
+/**
+ * Grabs referrer from v2 trackingMiddleware.
+ * In single-page-app context, value needs to refresh on route changes
+ * See: https://github.com/artsy/force/blob/master/src/v2/Artsy/Analytics/trackingMiddleware.ts
+ * @param options
+ */
+export const setAnalyticsClientReferrerOptions = (options: object = {}) => {
+  const referrer = window.analytics.__artsyClientSideRoutingReferrer
+  let trackingOptions = options
+
+  if (referrer) {
+    trackingOptions = {
+      page: {
+        referrer,
+      },
+    }
+  }
+  return trackingOptions
+}
+
```

Comment:
(thought) Since this is kind of a weird one that crosses through a few areas it might benefit from its own file